dipy/reconst/ivim.py

```diff
@@ -0,0 +1,541 @@
+""" Classes and functions for fitting ivim model """
+from __future__ import division, print_function, absolute_import
+
+from distutils.version import LooseVersion
+
+import numpy as np
+import scipy
+import warnings
+from dipy.reconst.base import ReconstModel
+from dipy.reconst.multi_voxel import multi_voxel_fit
+
+SCIPY_LESS_0_17 = LooseVersion(scipy.version.short_version) < '0.17'
+
+if SCIPY_LESS_0_17:
+    from scipy.optimize import leastsq
+else:
+    from scipy.optimize import least_squares
+
+
+def ivim_prediction(params, gtab, S0=1.):
+    """The Intravoxel incoherent motion (IVIM) model function.
+
+    Parameters
+    ----------
+    params : array
+        An array of IVIM parameters - [S0, f, D_star, D].
+
+    gtab : GradientTable class instance
+        Gradient directions and bvalues.
+
+    S0 : float, optional
+        This has been added just for consistency with the existing
+        API. Unlike other models, IVIM predicts S0 and this is over written
+        by the S0 value in params.
+
+    Returns
+    -------
+    S : array
+        An array containing the IVIM signal estimated using given parameters.
+    """
+    S0, f, D_star, D = params
+    b = gtab.bvals
+    S = S0 * (f * np.exp(-b * D_star) + (1 - f) * np.exp(-b * D))
+    return S
+
+
+def _ivim_error(params, gtab, signal):
+    """Error function to be used in fitting the IVIM model.
+
+    Parameters
+    ----------
+    params : array
+        An array of IVIM parameters - [S0, f, D_star, D]
+
+    gtab : GradientTable class instance
```

Comment:
Minor thought for @arokem and @sahmed95, the other things in gtab aren't being used, only the bvals. Maybe the input should just be the bvals?